airflow/models/taskinstance.py

```diff
@@ -1692,69 +2411,14 @@ def signal_handler(signum, frame):
         Stats.incr("operator_successes", tags={**self.stats_tags, "task_type": self.task.task_type})
         Stats.incr("ti_successes", tags=self.stats_tags)
 
-    def _run_finished_callback(
-        self,
-        callbacks: None | TaskStateChangeCallback | list[TaskStateChangeCallback],
-        context: Context,
-        callback_type: str,
-    ) -> None:
-        """Run callback after task finishes."""
-        if callbacks:
-            callbacks = callbacks if isinstance(callbacks, list) else [callbacks]
-            for callback in callbacks:
-                try:
-                    callback(context)
-                except Exception:
-                    callback_name = qualname(callback).split(".")[-1]
-                    self.log.exception(
-                        f"Error when executing {callback_name} callback"  # type: ignore[attr-defined]
-                    )
-
     def _execute_task(self, context, task_orig):
-        """Execute Task (optionally with a Timeout) and push Xcom results."""
-        task_to_execute = self.task
-        # If the task has been deferred and is being executed due to a trigger,
-        # then we need to pick the right method to come back to, otherwise
-        # we go for the default execute
-        execute_callable_kwargs = {}
-        if self.next_method:
```

Comment:
this is where the code from my comment above is coming from, there is only one check here

Reply:
Yeah. No harm but would be great to fix it. Care for opening PR?

Reply:
#38327